server/src/main/java/org/apache/druid/metadata/PendingSegmentRecord.java

```diff
@@ -40,7 +40,7 @@
  * <li>  id -> id (Unique identifier for pending segment) <li/>
  * <li>  sequence_name -> sequenceName (sequence name used for segment allocation) <li/>
  * <li>  sequence_prev_id -> sequencePrevId (previous segment id used for segment allocation) <li/>
- * <li>  upgraded_from_segment_id -> upgradedFromSegmentId (Id of the root segment from which this was upgraded) <li/>
+ * <li>  upgraded_from_segment_id -> upgradedFromSegmentId (Id of the first segment from which this was upgraded) <li/>
```

Comment:
```suggestion
 * <li>  upgraded_from_segment_id -> upgradedFromSegmentId (ID of the segment from which this was upgraded) <li/>
```

Reply:
I wanted to make it explicit that we are referring to the first or "root" segment in the case of multiple upgrades.

Reply:
Okay. For that, we should make it less ambiguous:

```
ID of the segment which was upgraded to create the current segment. If the former was itself created as a result of an upgrade, then this ID must refer to the original non-upgraded segment in the hierarchy.
```

Reply:
Done